### ISSUE TITLE
BUG : crash at import tome on python 3.9

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [beartype/beartype#406]
Hi,

here's the repro:

```
❯ pyenv virtualenv 3.9.0 test_beartype
Looking in links: /tmp/tmpnm56tu3_                                                       Requirement already satisfied: setuptools in /home/USER/.pyenv/versions/3.9.0/envs/test_beartype/lib/python3.9/site-packages (49.2.1)                                             
Requirement already satisfied: pip in /home/USER/.pyenv/versions/3.9.0/envs/test_beartype/lib/python3.9/site-packages (20.2.3)         
❯ pyenv activate test_beartype                                   
 ❯ python -m pip install beartype
                                  Collecting beartype
  Using cached beartype-0.18.5-py3-none-any.whl (917 kB)          Installing collected packages: beartype
Successfully installed beartype-0.18.5
WARNING: You are using pip version 20.2.3; however, version 24.1.1 is available.
You should consider upgrading via the '/home/USER/.pyenv/versions/test_beartype/bin/python -m pip install --upgrade pip' command.      ❯ python
Python 3.9.0 (default, Sep  8 2022, 15:30:10)
[GCC 11.2.0] on linux                                             Type "help", "copyright", "credits" or "license" for more information.
>>> import beartype
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/USER/.pyenv/versions/test_beartype/lib/python3.9/site-packages/beartype/__init__.py", line 58, in <module>
    from beartype._decor.decormain import (
  File "/home/USER/.pyenv/versions/test_beartype/lib/python3.9/site-packages/beartype/_decor/decormain.py", line 26, in <module>
    from beartype._conf.confcls import (
  File "/home/USER/.pyenv/versions/test_beartype/lib/python3.9/site-packages/beartype/_conf/confcls.py", line 46, in <module>
    from beartype._conf.confoverrides import (
  File "/home/USER/.pyenv/versions/test_beartype/lib/python3.9/site-packages/beartype/_conf/confoverrides.py", line 15, in <module>
    from beartype._data.hint.datahinttyping import (
  File "/home/USER/.pyenv/versions/test_beartype/lib/python3.9/site-packages/beartype/_data/hint/datahinttyping.py", line 290, in <module>
    BeartypeReturn = Union[BeartypeableT, BeartypeConfedDecorator]
  File "/home/USER/.pyenv/versions/3.9.0/lib/python3.9/typing.py", line 243, in inner
    return func(*args, **kwds)
  File "/home/USER/.pyenv/versions/3.9.0/lib/python3.9/typing.py", line 316, in __getitem__
    return self._getitem(self, parameters)
  File "/home/USER/.pyenv/versions/3.9.0/lib/python3.9/typing.py", line 421, in Union
    parameters = _remove_dups_flatten(parameters)
  File "/home/USER/.pyenv/versions/3.9.0/lib/python3.9/typing.py", line 215, in _remove_dups_flatten
    all_params = set(params)
TypeError: unhashable type: 'list'
```

**Comment by leycec** [comment]
*...heh.* The classic `TypeError: unhashable type: 'list'` exception, huh? Yeah. This isn't actually @beartype's fault. This is CPython's fault. Specifically, this is a bug in older implementations of the standard `typing` module under CPython < 3.9.2.

CPython 3.9.0 is over two years old and painfully broken. You desperately want to upgrade CPython. Of course, you desperately want to upgrade CPython *anyway.* Obsolete versions of anything are security risks that increase attack surface. Doubly so if the security risk in question is CPython itself. I grimace. :grimacing: 

## tl;dr

**Upgrade obsolete broken CPython versions.** Since this is a duplicate of long-closed issue #99, let's quietly close this for the sanity of this exploding issue tracker.

**Comment by thiswillbeyourgithub** [comment]
Oh I'm sorry I wasn't clear : the actual issue is more that pypi indicates that beartype can be used since python 3.8

Since I discovered beartype I tried to add it to all my previous repo and ran into this issue.

**Comment by leycec** [comment]
You're in luck! @beartype totally supports CPython ≥ 3.8. So, what's the catch? **@beartype doesn't support obsolete CPython patch releases that are broken.** @beartype can't support those, because those are broken. Sadly, both CPython 3.9.0 *and* CPython 3.9.1 are broken. Not our fault, thankfully. The standard `typing` module in both CPython 3.9.0 *and* CPython 3.9.1 has critical bugs. It just doesn't work... *at all.* Not much @beartype can do about that. It is sad. :sob:

The solution is just to update your CPython 3.9 to a **non-broken patch release.** For example, the current CPython patch release is **CPython 3.9.19.** Right? Please just upgrade to that. Then everything will work, @beartype will rejoice, and the Universe itself with dance with glee.

**tl;dr:** please just upgrade your CPython 3.9 interpreter to **CPython 3.9.19**. I guarantee worky.

**Comment by thiswillbeyourgithub** [comment]
Thanks a lot!
